mlx/io/load.h

```diff
@@ -170,5 +171,180 @@ class FileWriter : public Writer {
   std::string label_;
 };
 
+struct json;
+
+using json_array = std::vector<json>;
+using json_object = std::unordered_map<std::string, json>;
+
+struct json {
+  json() : data(json_object()) {}
+
+  template <typename T>
+  json(T x) : data(x) {}
+
+  // TODO: templatize this
+  json(std::vector<size_t> x) {
```

Comment:
Why not templatize these vector functions? Should be fairly straight-forward to remove the duplicates below?

```
template <typename T>
json(std::vector<T> x) {
...
```

Reply:
Oh I see, you have `long` for both `int` and `long` and `double` for bout `double` and `float` 🤔

Reply:
Totally agree -- for reasons I don't understand the compiler does not like it when I templatize here. I'll play around with it a little.